### ISSUE TITLE
🐛 SPARQL query demands period which spec says is optional

### ISSUE THREAD

**Issue by sroze** [linkeddata/rdflib.js#459]
I'm trying to use a SPARQL query to fetch from my store and it doesn't return any results. Here's a very simple reproducer.

## Reproducer 

```js
const $rdf = require('rdflib');
const store  = $rdf.graph();
store.add(
  store.sym('https://example.com/A'),
  store.sym('https://example.com/foo'),
  'bar',
);

const sparqlQuery = 'CONSTRUCT WHERE { ?nodeA ?edge ?nodeB }';
const query = $rdf.SPARQLToQuery(sparqlQuery, false, store);

store.query(query, result => {
  console.log(result);

  // Actual: []
  // Expected: The node added above 😢

});
```

**Comment by Melchyore** [comment]
Hello!

I have the same problem here! Queries always return an empty array! :(

**Comment by jeff-zucker** [comment]
@Melchyore  rdflib has its own take on SQL.  One of the things is that it demands a period after statements.  Put a period after "?nodeB" in the query above and you'll get data back as expected.

**Comment by jeff-zucker** [comment]
So, I'm going to close this issue, because it was a syntax error in the use case, not a bug.

**Comment by TallTed** [comment]
@jeff-zucker -- I disagree that `rdflib` treating a period as required, where it is optional according to SPARQL spec, is not a bug.

Anywhere `rdflib` evinces "its own take on SPARQL" (certainly anywhere `rdflib` is more stringent, and arguably anywhere `rdflib` is more forgiving) is absolutely a bug in `rdflib`, and this requirement by `rdflib` should be addressed in its code.

Please reopen this issue.

**Comment by jeff-zucker** [comment]
Fair enough, reopened with better title. I wasn't able to find the period mentioned in the spec, can you point me to it?  The examples at W3 are inconsistent, sometimes showing a period and sometimes not.

**Comment by TallTed** [comment]
I don't know whether the optionality is ever discussed in the prose. I can't quickly find it.  However --

> [Section 19](https://www.w3.org/TR/sparql11-query/#grammar) contains the normative definition of the syntax for the SPARQL query and [SPARQL update](http://www.w3.org/TR/sparql11-update/) languages, as given by a grammar expressed in EBNF notation.

-- and this comes from that Section 19 --

```
[52]     TriplesTemplate     ::= TriplesSameSubject ( '.' TriplesTemplate? )?
```

Note that the `'.'` is contained in an element that appears zero or more times (expressed by the trailing `?`).  Hopefully that's sufficient....
